src/setup/txt_joybinput.c

```diff
@@ -381,6 +574,93 @@ static void TXT_JoystickInputDrawer(TXT_UNCAST_ARG(joystick_input))
     }
 }
 
+static int GetGamepadButtonIndex(int button)
+{
+    int i;
+
+    for (i = 0; i < arrlen(gamepad_buttons); ++i)
+    {
+        if (button == gamepad_buttons[i])
+        {
+            return i;
+        }
+    }
+
+    I_Error("Couldn't find button index.");
+    return -1;
```

Comment:
`I_Error` does not return.

Reply:
Instead of calling `I_Error`, I will display a button name of "(unknown)".